core/src/main/scala/org/apache/spark/deploy/DependencyUtils.scala

```diff
@@ -137,16 +137,29 @@ private[deploy] object DependencyUtils {
   def resolveGlobPaths(paths: String, hadoopConf: Configuration): String = {
     require(paths != null, "paths cannot be null.")
     Utils.stringToSeq(paths).flatMap { path =>
-      val uri = Utils.resolveURI(path)
-      uri.getScheme match {
-        case "local" | "http" | "https" | "ftp" => Array(path)
-        case _ =>
-          val fs = FileSystem.get(uri, hadoopConf)
-          Option(fs.globStatus(new Path(uri))).map { status =>
-            status.filter(_.isFile).map(_.getPath.toUri.toString)
-          }.getOrElse(Array(path))
+      val spath = path.split('#')
+      val renameAs = if (spath.length > 1) Some(spath(1)) else None
+      val resolved: Array[String] = resoloveGlobPath(spath(0), hadoopConf)
+      resolved match {
```

Comment:
This can be simplified something like this: `(renameAs, resolved) match...`

Reply:
This whole match block is a little ugly, but I'll wait to see how you implement Gabor's suggestion...